File: 1763-longest-nice-substring/1763-longest-nice-substring.java
class Solution {
    public String longestNiceSubstring(String s) {
        String result = "";
           
        for (int i = 0; i < s.length(); i++) {
            boolean[] lower = new boolean[26];
            boolean[] upper = new boolean[26];
            
            for (int j = i; j < s.length(); j++) {
                if (s.charAt(j) < 'a') upper[s.charAt(j) - 'A'] = true;
                else                   lower[s.charAt(j) - 'a'] = true;
                
                if (Arrays.equals(lower, upper) && result.length() <= j - i) {
                    result = s.substring(i, j + 1);
                }                
            }
        }
        return result;
    }
}

File: 1763-longest-nice-substring/1763-longest-nice-substring.py
class Solution:
    def longestNiceSubstring(self, s: str) -> str:
        for i, c in enumerate(s): 
            if c.swapcase() not in s: 
                return max(self.longestNiceSubstring(s[:i]), self.longestNiceSubstring(s[i + 1:]), key=len)
        return s